### ISSUE TITLE
Ensures that CJK and English or numbers are separated by a single space

### ISSUE THREAD

**Issue by cworld1** [rvben/rumdl#843]
### Problem or motivation

It maybe a little hard to achieve. And I just want to know if this can be taken into consideration.

This feature actually is from the obsidian linter plugin [Space between Chinese Japanese or Korean and English or numbers](https://platers.github.io/obsidian-linter/settings/spacing-rules/#space-between-chinese-japanese-or-korean-and-english-or-numbers)

### Proposed solution

Here is the guideline: [Chinese copywriting guidelines for better written communication](https://github.com/sparanoid/chinese-copywriting-guidelines)

### Alternatives considered

Appreciate if this can be considered.

### Feature category

New lint rule

### Example markdown

Incorrect:

```markdown
日本語englishひらがな
カタカナenglishカタカナ
ﾊﾝｶｸｶﾀｶﾅenglish１２３全角数字
한글english한글
```

Correct:

```markdown
日本語 english ひらがな
カタカナ english カタカナ
ﾊﾝｶｸｶﾀｶﾅ english１２３全角数字
한글 english 한글
```

### Contribution

- [ ] I would be willing to submit a PR to implement this feature

**Comment by rvben** [comment]
Thanks for suggesting this, and especially for the guidelines link and the before/after examples. They are what the rule was written against, so they shaped it directly.

Shipped as **MD089 (`cjk-spacing`)** in v0.2.64. It is opt-in, since technical Chinese writing often prefers no space:

```toml
extend-enable = ["MD089"]
```

Running `rumdl fmt` against your "Incorrect" block produces your "Correct" block byte for byte, including `ﾊﾝｶｸｶﾀｶﾅ english１２３全角数字`, where the fullwidth digits correctly get no space. The symbol handling is configurable via `symbols-after-cjk` / `symbols-before-cjk`; see `docs/md089.md`.

**Comment by cworld1** [comment]
i just thought it was a rather difficult thing and never expected it to come true. appreciate for your hard working! ❤️❤️❤️

**Comment by rvben** [comment]
Thanks @cworld1 for the nice words. If you hit anything else with CJK text, let me know. 😁
